### PR TITLE
Fix 'tab' command not working with TabPane

### LINKED ISSUE #112: 'tab' command does not update the contents of the tab properly
This is due to the change from using menu bar to tab pane in #101.

### PR DESCRIPTION
**Pull request by ianyong**
Changes:
- Update the 'tab' command to change the tab via the `TabPane`.
- Add tab index information to the `UiState.Tab` enum.
- Update corresponding tests.

Fixes #112.

### CODE REVIEW

**Review by yongping827 — DISMISSED** [review]
LGTM

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/UiState.java [hunk 1]
```diff
@@ -10,10 +10,31 @@
...
+
+        /** The index of the tab in the {@code TabPane}. */
+        private final int tabIndex;
+
+        /**
+         * Constructs a new {@code Tab} enum with the specified tab index.
+         *
+         * @param tabIndex The index of the tab in the {@code TabPane}.
+         */
+        private Tab(int tabIndex) {
```
Comment (wltan) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/UiState.java:26]: image

Does this need to be fixed?

**Review by wltan — APPROVED** [review]
LGTM

**Review by siddarth2824 — APPROVED** [review]
LGTM!

**Pull request merged**